.github/workflows/examples-nrfconnect.yaml

```diff
@@ -39,6 +39,8 @@ jobs:
               with:
                   fetch-depth: 0
                   submodules: true
+            - name: Update nRF Connect SDK revision to the currently recommended.
```

Comment:
This way seems easier to manage to me. Why put source code into docker?

Reply:
We generally update docker quite regularly, but we sometimes need to update it more frequently (e.g. while working on SRP/mDNS we need access to the most recent OpenThread features) and updating the Docker becomes a bottleneck. Updating nRF Connect SDK by a few commits shouldn't take so long. We can also look for some savings in CI to compensate the overhead, like not pulling all submodules, but only those which are needed. Would that be OK with you?

Reply:
Note that updating the SDK version is more less like fetching and switching to a different commit in a repository. It's not pulling all the SDK and fully replacing the existing one.

Reply:
Is it actually faster to put the repos into docker vs clone with depth 1?

Reply:
I think it is, because shallow clone is not yet supported in west (the tool used by Zephyr/nRF Connect to manage all involved git repositories): https://github.com/zephyrproject-rtos/west/issues/319. A lot of people are missing the feature, so possibly it will be done soon.

Reply:
One CI optimization idea that we should definitely consider is removing ARM gcc from the Docker images since it comes with Pigweed, anyway.

Reply:
The oddity I found was having a base image in docker that gets updated at CI time. Recent woes with cirque running for 30 minutes makes me weary of this (cirque also does a 'bootstrap' at CI time because docker image is too old).

Ideally I would either have the thing in docker (with matching instructions on how to update when not using docker) or automated retreive (like we do git submodules). A mix of both seems more complex to me.

Gave the checkmark in any case. figured we can have some GH issue describing that more standardization could help.